src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,143 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * The constructor of the class
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Marks a task as done
+     *
+     * @param num Specifies which task to mark
+     * @param ui  An object to print output
+     */
+    public void mark(int num, Ui ui) {
+        tasks.get(num - 1).mark();
+        ui.printMessage("OK, I've marked this task as done:");
+        ui.printMessage(tasks.get(num - 1).toString());
+    }
+
+    /**
+     * Unmarks a task
+     *
+     * @param num Specifies which task to unmark
+     * @param ui  An object to print output
+     */
+    public void unmark(int num, Ui ui) {
+        tasks.get(num - 1).unMark();
+        ui.printMessage("OK, I've marked this task as not done yet:");
+        ui.printMessage(tasks.get(num - 1).toString());
+    }
+
+    /**
+     * Deletes a task
+     *
+     * @param num Specifies which item to delete
+     * @param ui  An object to print output
+     */
+    public void delete(int num, Ui ui) {
+        Task removedTask = tasks.get(num - 1);
+        tasks.remove(num - 1);
+        ui.printMessage("Noted. I've removed this task:" + "\n" + removedTask.toString()
+                + "Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    /**
+     * Adds a task
+     *
+     * @param item The String that needs to be parsed and add to the list
+     * @throws DukeException Throws an exception
+     */
+    public void addATask(String item) throws DukeException {
+        String[] arr = item.split(" ", 2);
+        String type = arr[0];
+        switch (type) {
```

Comment:
This is not a code quality violation, but rather a coding standard violation. In CS2103T, if I remember correctly, the cases should not have an additional indent with respect to the switch line.